<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Read a sequence index — read_seqs • gggenomes</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  <script src="../extra.js"></script>

<meta property="og:title" content="Read a sequence index — read_seqs" />
<meta property="og:description" content="Read ID, description and length for each sequence from common formats
including FASTA, samtools/seqkit FASTA index files, and GFF3. Default columns
are &quot;seq_id&quot;, &quot;seq_desc&quot; and &quot;length&quot;." />
<meta property="og:image" content="/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gggenomes</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/emales.html">From a few sequences to a complex map in minutes</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/thackl/gggenomes/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read a sequence index</h1>
    <small class="dont-index">Source: <a href='https://github.com/thackl/gggenomes/blob/master/R/read_seqs.R'><code>R/read_seqs.R</code></a></small>
    <div class="hidden name"><code>read_seqs.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Read ID, description and length for each sequence from common formats
including FASTA, samtools/seqkit FASTA index files, and GFF3. Default columns
are "seq_id", "seq_desc" and "length".</p>
    </div>

    <pre class="usage"><span class='fu'>read_seqs</span><span class='op'>(</span><span class='va'>files</span>, format <span class='op'>=</span> <span class='cn'>NULL</span>, .id <span class='op'>=</span> <span class='st'>"file_id"</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='fu'>read_seq_len</span><span class='op'>(</span>
  <span class='va'>file</span>,
  col_names <span class='op'>=</span> <span class='fu'><a href='def_names.html'>def_names</a></span><span class='op'>(</span><span class='st'>"seq_len"</span><span class='op'>)</span>,
  col_types <span class='op'>=</span> <span class='fu'><a href='def_types.html'>def_types</a></span><span class='op'>(</span><span class='st'>"seq_len"</span><span class='op'>)</span>,
  <span class='va'>...</span>
<span class='op'>)</span>

<span class='fu'>read_fai</span><span class='op'>(</span><span class='va'>file</span>, col_names <span class='op'>=</span> <span class='fu'><a href='def_names.html'>def_names</a></span><span class='op'>(</span><span class='st'>"fai"</span><span class='op'>)</span>, col_types <span class='op'>=</span> <span class='fu'><a href='def_types.html'>def_types</a></span><span class='op'>(</span><span class='st'>"fai"</span><span class='op'>)</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>file</th>
      <td><p>fasta or .fai/.seqkit.fai fasta index</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>gggenomes-compatible seqs tibble</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># from a fasta file</span>
<span class='fu'>read_seqs</span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales.fna"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading as seq_len:</span></div><div class='output co'>#&gt; <span class='message'>* emales [/home/runner/work/_temp/Library/gggenomes/extdata/emales/emales.fna]</span></div><div class='output co'>#&gt; <span style='color: #555555;'># A tibble: 0 x 1</span><span>
#&gt; </span><span style='color: #555555;'># … with 1 variable: file_id </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span></div><div class='input'><span class='co'># from samtools/seqkit style index</span>
<span class='fu'>read_seqs</span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales.fna.seqkit.fai"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading as fai:</span></div><div class='output co'>#&gt; <span class='message'>* emales.fna.seqkit [/home/runner/work/_temp/Library/gggenomes/extdata/emales/emales.fna.seqkit.fai]</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 6 x 4</span><span>
#&gt;   file_id          seq_id     seq_desc                                    length
#&gt;   </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span>                                        </span><span style='color: #555555;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #555555;'>1</span><span> emales.fna.seqk… BVI_008A   emale_type=EMALE01 is_typespecies=FALSE ha…  </span><span style='text-decoration: underline;'>26</span><span>820
#&gt; </span><span style='color: #555555;'>2</span><span> emales.fna.seqk… BVI_069    emale_type=EMALE01 is_typespecies=FALSE ha…  </span><span style='text-decoration: underline;'>26</span><span>808
#&gt; </span><span style='color: #555555;'>3</span><span> emales.fna.seqk… Cflag_017B emale_type=EMALE01 is_typespecies=TRUE has…  </span><span style='text-decoration: underline;'>21</span><span>311
#&gt; </span><span style='color: #555555;'>4</span><span> emales.fna.seqk… E4-10_086  emale_type=EMALE01 is_typespecies=FALSE ha…  </span><span style='text-decoration: underline;'>20</span><span>642
#&gt; </span><span style='color: #555555;'>5</span><span> emales.fna.seqk… E4-10_112  emale_type=EMALE01 is_typespecies=FALSE ha…  </span><span style='text-decoration: underline;'>26</span><span>856
#&gt; </span><span style='color: #555555;'>6</span><span> emales.fna.seqk… RCC970_01… emale_type=EMALE01 is_typespecies=FALSE ha…  </span><span style='text-decoration: underline;'>20</span><span>152</div><div class='input'><span class='co'># from multiple gff file</span>
<span class='fu'>read_seqs</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales.gff"</span><span class='op'>)</span>, <span class='fu'><a href='ex.html'>ex</a></span><span class='op'>(</span><span class='st'>"emales/emales-tirs.gff"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Reading as seq_len:</span></div><div class='output co'>#&gt; <span class='message'>* emales [/home/runner/work/_temp/Library/gggenomes/extdata/emales/emales.gff]</span></div><div class='output co'>#&gt; <span class='message'>* emales-tirs [/home/runner/work/_temp/Library/gggenomes/extdata/emales/emales-tirs.gff]</span></div><div class='output co'>#&gt; </span><span style='color: #555555;'># A tibble: 0 x 1</span><span>
#&gt; </span><span style='color: #555555;'># … with 1 variable: file_id </span><span style='color: #555555;font-style: italic;'>&lt;chr&gt;</span><span></div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Thomas Hackl, Markus J. Ankenbrand.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


